src/NumericalAlgorithms/Interpolation/TryToInterpolate.hpp

```diff
@@ -78,36 +116,75 @@ void interpolate_data(
-                db::AddSimpleTags<::Tags::Variables<
-                    typename Metavariables::interpolator_source_vars>>,
-                db::AddComputeTags<
-                    typename InterpolationTargetTag::compute_items_on_source>>(
-                volume_info.vars);
-
-            Variables<
-                typename InterpolationTargetTag::vars_to_interpolate_to_target>
-                local_vars(volume_info.mesh.number_of_grid_points());
-
-            tmpl::for_each<
-                typename InterpolationTargetTag::vars_to_interpolate_to_target>(
-                [&new_box, &local_vars](auto x) noexcept {
-                  using tag = typename decltype(x)::type;
-                  get<tag>(local_vars) = db::get<tag>(new_box);
-                });
+            auto& volume_info = volume_info_outer.second.at(element_id);
+            auto& vars_to_interpolate =
+                get<::intrp::Tags::VarsToInterpolateToTarget<
+                    InterpolationTargetTag>>(volume_info.vars_to_interpolate);
+
+            if constexpr (detail::has_compute_vars_to_interpolate_v<
+                              InterpolationTargetTag>) {
+              if (vars_to_interpolate.size() == 0) {
+                // vars_to_interpolate has not been filled for
+                // this element at this temporal_id.  So fill it. How we
+                // fill it will depend on whether we need to change frames.
+                vars_to_interpolate.initialize(
+                    volume_info.vars_from_element.number_of_grid_points());
+
+                if constexpr (any_index_in_frame_v<typename Metavariables::
+                                                       interpolator_source_vars,
+                                                   Frame::Inertial> and
+                              any_index_in_frame_v<
+                                  typename InterpolationTargetTag::
+                                      vars_to_interpolate_to_target,
+                                  Frame::Grid>) {
+                  // Need to do frame transformations.
+
+                  // The functions of time are always guaranteed to be
+                  // up-to-date here, because they are guaranteed to be
+                  // up-to-date before calling SendPointsToInterpolator
+                  // (which is guaranteed to be called before
+                  // interpolate_data is called).
+                  const auto& functions_of_time =
+                      get<domain::Tags::FunctionsOfTime>(cache);
+                  const auto& block = domain.blocks().at(element_id.block_id());
+                  ElementMap<3, ::Frame::Grid> map_logical_to_grid{
+                      element_id,
+                      block.moving_mesh_logical_to_grid_map().get_clone()};
+                  const auto invjac_logical_to_grid =
+                      map_logical_to_grid.inv_jacobian(
+                          logical_coordinates(volume_info.mesh));
```

Comment:
Would it make sense to send the Jacobians, rather than recompute them here?

Reply:
Downsides to sending them:
  - More data sent.
  - Would need to store jacobians at multiple temporal_ids.
  - Not sure how the interface would change.  You would need to send Jacobians only when interpolating onto non-inertial-frames, so the easiest way to do it would require you to specify sending Jacobians in the Metavariables, but only for interpolating to some frames.

Upsides to sending them:
 - Don't need to compute them here.
 
 So I guess the question is how expensive is it to compute Jacobians?  If it's cheap, then better to recompute. If it's expensive, then better to send.

Reply:
OK.  I don't have a good feel for how expensive any of these things are, so we can change it later if it becomes an issue.